snare.conf.5

```diff
@@ -225,11 +256,11 @@ github {
   reposdir = "<path>";
   match ".*" {
     cmd = "/path/to/prps/%o/%r %e %j";
-    email = "abc@def.com";
+    error_cmd = "cat %s | mailx -s \\"snare error: github.com/%o/%r\\" abc@def.com";
     secret = "sec";
   }
   match "a/b" {
-    email = "ghi@jkl.com";
+    error_cmd = "xyz";
```

Comment:
Shall we make that a real command to demonstrate another use of errorcmd?

Reply:
I would love to, but I failed to think of a good example :( Suggestions are very welcome!

Reply:
Most of the actually useful ones are going to be quite involved, so how about something fun?

```
lp %s
```

```
notify-send "It's broken Dave" "`cat %s`"
```

Reply:
What's `lp`? `notify-send` I considered, but it seemed unlikely that someone running a daemon like snare would do so on a desktop!

Reply:
What's `lp`? That's how you used to print things to your dot-matrix printer back in 1982 :)

> notify-send I considered, but it seemed unlikely that someone running a daemon like snare would do so on a desktop

Yeah, but it's just for fun! A command that a reader recognises must be better than `xyz`.

If you wanted something more serious, you could append the error to a file some place.

Reply:
Well, we could do `echo "Error in %o/%r" | lpr -` which I think will work on a modern machine?

Reply:
`lpr` is fine.

Wouldn't you need a `%s` though? I thought snare put stderr/stdout to file?

Reply:
You don't *have* to use any of the `%` modifiers if you don't want to. But yeah `lrp %s` has the virtue of simplicity!

Reply:
I'm missing some detail about stdout/stderr.

So if `%s` is absent, it pushes the error text to stdin of the errorcmd?

Reply:
Fixed in https://github.com/softdevteam/snare/pull/65/commits/ffa53cf2c1095d001bb82b8d69b6aa0e3c7b90df.

Reply:
> I'm missing some detail about stdout/stderr.

Have you read the man page? Does that explain it adequately?

Reply:
I misread.

You are proposing: `echo "Error in %o/%r" | lpr -`, which doesn't include the error text at all.

So OK.